microprofile-config-keyvault/src/main/java/com/azure/microprofile/config/keyvault/AzureKeyVaultConfigSource.java

```diff
@@ -8,16 +8,43 @@
 import java.util.Map;
 import java.util.Set;
 
+/**
+ * Custom ConfigSource for Azure Key Vault.
+ */
 public class AzureKeyVaultConfigSource implements ConfigSource {
 
     private AzureKeyVaultOperation keyVaultOperation;
 
     private boolean isKeyVaultEnabled = true;
 
+    /**
+     * Default constructor.
+     */
     public AzureKeyVaultConfigSource() {
         // no op
     }
 
+    /**
+     * Initialize {@link AzureKeyVaultOperation} based on configuration.
+     *
+     * <ul>
+     *  <li>If {@code azure.keyvault.url} is not set, then {@link AzureKeyVaultOperation} will not be initialized.</li>
+     *  <li>
+     *      If {@code azure.keyvault.cache} is set to {@code true}, then {@link CachedAzureKeyVaultOperation} will be used.
+     *      <ul>
+     *          <li>If {@code azure.keyvault.cache.ttl} is set, then it will be used as TTL for cache entries.</li>
+     *          <li>Otherwise, {@link CachedAzureKeyVaultOperation#DEFAULT_CACHE_REFRESH_INTERVAL_IN_MS} will be used as TTL for cache entries.</li>
+     *      </ul>
+     *  </li>
+     *  <li>
+     *      Otherwise, {@link NoCacheAzureKeyVaultOperation} will be used.
+     *      <ul>
+     *          <li>If {@code azure.keyvault.secret-name-regex} is set, then it will be used to filter secret names.</li>
+     *          <li>Otherwise, {@link NoCacheAzureKeyVaultOperation#DEFAULT_SECRET_NAME_REGEX} will be used to filter secret names.</li>
+     *      </ul>
+     *  </li>
+     * </ul>
+     */
```

Comment:
## Questions
1. Can we just call the init() once?
2. Can we log here if the value of isKeyVaultEnabled is false
<img width="612" alt="image" src="https://github.com/Azure/azure-microprofile/assets/4465723/4b1849a6-aada-4aa5-9633-4ad7160df1fb">

Reply:
> Can we just call the init() once?

Nice comment. I re-visited the code and it turns out that `init()` can be invoked just once in the constructor. Addressed in [2a8ca00](https://github.com/Azure/azure-microprofile/pull/7/commits/2a8ca00bff508bd8e2d942c511aade1da48c3cdc).

> Can we log here if the value of isKeyVaultEnabled is false

Sure. Fixed in [7ff5098](https://github.com/Azure/azure-microprofile/pull/7/commits/7ff5098a23ed519beb857d59d3954b1ff77a0684).